### PR TITLE
Add Thank you Modal for survey submission

### PR DESCRIPTION
**Pull request by djnunez-aot**
*Issue #421 :*
https://github.com/bcgov/met-public/issues/#421

Add Thank you modal for survey submission

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

File: met-web/src/components/survey/submit/SurveyForm.tsx [hunk 1]
```diff
@@ -1,52 +1,82 @@
...
+        handleSubmit(submissionData);
+    };
+
     if (isLoading) {
         return <Skeleton variant="rectangular" height="50em" width="100%" />;
     }
 
+    const handleModalClose = () => {
+        setModalVisible(false);
+        navigate('/');
```
Comment (jadmsaadaot) [inline met-web/src/components/survey/submit/SurveyForm.tsx:35]: navigate public users to '/'?

File: met-web/src/components/survey/submit/SurveyForm.tsx [hunk 2]
```diff
@@ -1,52 +1,82 @@
...
-                    </PrimaryButton>
-                </Stack>
+        <>
+            <Modal
+                open={modalVisible}
+                onClose={() => handleModalClose()}
+                aria-labelledby="modal-modal-title"
+                aria-describedby="modal-modal-description"
+            >
+                <SuccessPanel
```
Comment (jadmsaadaot) [inline met-web/src/components/survey/submit/SurveyForm.tsx:46]: Please  review with MA, I believe she wanted this modal to appear on the Engagement View page after the user is redirected when they do a successful survey and not on the survey submit page itself.

File: met-web/src/components/survey/submit/SurveyForm.tsx [hunk 3]
```diff
@@ -1,52 +1,82 @@
...
     const [isValid, setIsValid] = useState(false);
+    const [modalVisible, setModalVisible] = useState(false);
 
     const handleChange = (filledForm: FormSubmissionData) => {
         setSubmissionData(filledForm.data);
         setIsValid(filledForm.isValid);
     };
 
+    const submit = (submissionData: unknown) => {
+        setModalVisible(true);
```
Comment (jadmsaadaot) [inline met-web/src/components/survey/submit/SurveyForm.tsx:25]: so automatically success modal is shown, what if handleSubmit failed?

**Review by jadmsaadaot — CHANGES_REQUESTED** [review]
Review with Marie-Anik whether she wants this success modal on survey page or engagement page

File: met-web/src/components/survey/submit/SurveyForm.tsx (hunk 1 above)
Comment (tiagograf-AOT) [reply met-web/src/components/survey/submit/SurveyForm.tsx:35]: this should probably be engagement view

**Pull request closed without merging**